Source/DataSources/KmlDataSource.js

```diff
@@ -1947,20 +1945,52 @@ define([
         Tour : processUnsupported
     };
 
+    var promisifiedFeatureTypes = {
+        // already promisified
+        Document : processDocument,
+        // already promisified
+        Folder : processFolder,
+        Placemark : deferAndWrapInPromise(processPlacemark),
+        NetworkLink : deferAndWrapInPromise(processNetworkLink),
+        GroundOverlay : deferAndWrapInPromise(processGroundOverlay),
+        PhotoOverlay : deferAndWrapInPromise(processUnsupported),
+        ScreenOverlay : deferAndWrapInPromise(processUnsupported),
+        Tour : deferAndWrapInPromise(processUnsupported)
+    };
+
+    function deferAndWrapInPromise(originalFunction){
+        return function(){
+            var deferred = when.defer();
+            var args = Array.prototype.slice.call(arguments);
+            var context = this;
+            setTimeout(function deferredFunc(){
```

Comment:
Using `setTimeout` here is not an ideal solution because it happens outside the requestAnimationFrame event loop (and therefore has negative impact on overall performance).  In general, `setTimeout` should be avoided in any Cesium application for this reason.

Instead, we have `DataSource.update`, which is an optional function that will be called every frame during the event loop.  I think what we would want is incremental loading to happen by doing a little bit of KML work in every call to update (either through time-boxing or other form of work counting).  This approach may not lend itself well to the Promise-based approach (I'd have to think about it some more), but would be a much better solution (and is something we should probably due in all of our Datasources.  If you add `KmlDataSource.prototype.update` it should automatically start getting called every frame.

Perhaps we just change the `processXXX` functions to always return a "next" function that we can call to continue processing.  In update, we would call the next function as many times as possible in some pre-determined allotted time.  I don't think it should be hard to do.  What do you think?

@shunter any thoughts here?